topics/introduction/tutorials/python-advanced-np-pd/tutorial.md

```diff
@@ -0,0 +1,988 @@
+    a. 
+
+    ```python
+     data['P-value']
+	```
+    b.
+    ```python
+       data.loc['FBgn0039178', :]
+    ```
+    c. 
+    ```python
+       data[data['Chromosome'] == 'chr3R']
+    ```
+
+## Group-by and analyze the data
+
+Many data analysis tasks can be approached using the “split-apply-combine” paradigm: split the data into groups, apply some analysis to each group, and then combine the results.
+
+Pandas makes this very easy through the use of the `groupby()` method, which splits the data into groups. When the data is grouped in this way, the aggregate method `agg()` can be used to apply an aggregating or summary function to each group.
+
+```python
+summarised_data = data.groupby('Chromosome').agg({'Base mean':'first', 
+                             'log2(FC)': 'max'})
+print(summarised_data)
+```
+```output
+              Base mean  log2(FC)
+Chromosome                       
+chr2L       1269.362573  2.145440
+chr2R        928.263812  2.407986
+chr3L       5430.067277  2.428607
+chr3R       1086.974295  2.360017
+chrX        6409.577128  2.699939
+```
+
+There are a couple of things that should be noted. The `agg()` method accepts a dictionary as input that specifies the function to be applied to each column. The output is a new dataframe, that each row corresponds to one group. The output dataframe uses the grouping column as index. We could change the last one by simply using the `reset_index()` method.
+
+```python
+summarised_data = data.groupby('Chromosome').agg({'Base mean':'first', 
+                                                  'log2(FC)': 'max'}).reset_index() 
+print(summarised_data)
+```
+```output
+  Chromosome    Base mean  log2(FC)
+0      chr2L  1269.362573  2.145440
+1      chr2R   928.263812  2.407986
+2      chr3L  5430.067277  2.428607
+3      chr3R  1086.974295  2.360017
+4       chrX  6409.577128  2.699939
+```
+### Exercises
+
+1. Import the tabular data from the "https://zenodo.org/record/3477564/files/annotatedDEgenes.tabular" link. What are the longest genes in each chromosome?
+2. Using the same dataset,  try to find how many genes are found on each strand of each chromosome.
+
+
+### Solutions
+1. 
+    ```python
+    data = pd.read_csv("https://zenodo.org/record/3477564/files/annotatedDEgenes.tabular", sep = "\t", index_col = 'GeneID')
+    
+    data['Gene Length'] = data['End'] - data['Start']
+    data.groupby('Chromosome').agg(max_length = ('Gene Length', 'max'))
+    ```
+2. You can group the data according to more than one column.
+    ```python
+    data.groupby(['Chromosome', 'Strand']).size()
+    ```
+
+
+
+# Conclusion
+
+This tutorial aims to serve as an introduction to data analysis using the Python programming language.
+
+## Key points
+
+* Python has many libraries offering a variety of capabilities, which makes it popular for beginners, as well as, more experienced users
+
+* You can use scientific libraries like Numpy and Pandas to perform data analysis.
+
+
+
+# References
```

Comment:
same here, all of these should go. And citation statements should be used in the appropriate place

Reply:
So, I should remove only the references part or the conclusions and key points as well?

Reply:
Just the references, the rest is ok!